### PR TITLE
0038 Refactor to add item start date to the selected day

### PR DESCRIPTION
**Pull request by realityexpander**
* Refactored Event, Reminder & Task to use the selected day as default start date for new items
* Add real-time animation of time needle
* Add progress for in-progress events (yes not in spec, but useful)

### CODE REVIEW

File: app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt [hunk 1]
```diff
@@ -334,6 +339,23 @@ fun AgendaScreenContent(
...
+        }
+    } }
+    DisposableEffect(true) {
+        timer.scheduleAtFixedRate(timerTask, 0, 1000)
+
+        onDispose {
+            timerTask.cancel()
+            timer.cancel()
+        }
+    }
```
Comment (philipplackner) [inline app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt:357]: Any reason you didn't implement this UI state logic in the VM?

File: app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/components/AgendaCard.kt [hunk 2]
```diff
@@ -173,6 +177,24 @@ fun AgendaCard(
...
             )
+
+            // • Progress indicator for (for Events only)
+            toDateTime?.let {
+                AnimatedVisibility(zonedDateTimeNow.isAfter(fromDateTime) && zonedDateTimeNow.isBefore(toDateTime)) {
+                    Text(
+                        text = "In Progress: " +
+                                (100 - ( ( (toDateTime.toEpochSecond() - zonedDateTimeNow.toEpochSecond()).toFloat()
+                                  / (toDateTime.toEpochSecond() - fromDateTime.toEpochSecond()).toFloat() )*100))
+                                    .roundToInt() + "% Complete",
```
Comment (philipplackner) [inline app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/components/AgendaCard.kt:188]: Introducing some variables here would make this a lot more readable

File: app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt (hunk 1 above)
Comment (realityexpander) [reply app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt:357]: Im not sure what you mean here... This is a ticker to refresh the list and time needle. 

How should the VM do this? Isnt this the responsibility of the UI layer?

File: app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/components/AgendaCard.kt (hunk 2 above)
Comment (realityexpander) [reply app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/components/AgendaCard.kt:188]: ok will do.

File: app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt (hunk 1 above)
Comment (philipplackner) [reply app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt:357]: In the end you're updating UI state here what is usually the responsibility of the ViewModel unless there's a good reason to do it in the UI. The same could be done in the ViewModel with a timer which you cancel in `onCleared()`

Comment (realityexpander) [reply app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt:357]: OK I have refactored for this.

**Pull request merged**